### PR TITLE
Fix duplicate Urlbuilder

### PR DESCRIPTION
**Pull request by nklerk**
-Added Urlbuilder.js with function buildBrainUrl(Brain, url)
-input types can be object and string for brain and the url can be passed but isn't required.
-Removed functions named buildBrainUrl.
-Changed the code that used the buildBrainUrl function to use Urlbuilder.buildBrainUrl.
-Added tests to test the code.

### CODE REVIEW

File: test/unit/lib/device/brain/urlbuilder_test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const expect = require('chai').expect;
+const urlbuilder = require('../../../../../lib/device/brain/urlbuilder.js');
+const BASE_URL_GETRECIPES = '/v1/api/recipes';
+
+describe('./lib/device/brain/urlbuilder.js', function() {
+
+  it('should return brain url when object is provided', function(done) {
```
Comment (neophob) [inline test/unit/lib/device/brain/urlbuilder_test.js]: the done argument is only needed for async tests, they can all removed here

File: lib/device/brain/urlbuilder.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const DEFAULT_BRAIN_PORT = 3000;
+const PROTOCOL = 'http://';
+
+module.exports.buildBrainUrl = function(brain, baseUrl) {
+  if (!baseUrl){baseUrl='';}
```
Comment (neophob) [inline lib/device/brain/urlbuilder.js]: minor style issue, should look like

```
if (!baseUrl){
  baseUrl='';
}
```

File: test/unit/lib/device/brain/urlbuilder_test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    expect(test).to.equal('http://test.local:6336/v1/api/recipes');
+    done();
+  });
+
+  it('should return brain url and path when string is provided with uri', function(done) {
+    const test = urlbuilder.buildBrainUrl('brain.local', BASE_URL_GETRECIPES);
+    expect(test).to.equal('http://brain.local:3000/v1/api/recipes');
+    done();
+  });
+
```
Comment (neophob) [inline test/unit/lib/device/brain/urlbuilder_test.js]: another useful test would be, if we call `urlbuilder.buildBrainUrl()` -> then an error should be thrown.

the test should look like

```
    expect(function() {
      urlbuilder.buildBrainUrl();
    }).to.throw(/INVALID_PARAMETER/);
```

of course you need to add a sanity check for the buildBrainUrl function

**Comment by neophob** [comment]
Thanks @nklerk for this PR! some minor issues are open, as soon as they are resolved I'm happy to merge. 

and the test commit can be removed as well I guess

**Comment by nklerk** [comment]
Changes are made.

**Comment by neophob** [comment]
thanks!

**Pull request merged**